### PR TITLE
Can use image header in blog posts and images in post content

### PR DESCRIPTION
**Pull request by LikkanChung**

### CODE REVIEW

File: _layouts/post.html [hunk 1]
```diff
@@ -16,9 +16,21 @@
   {% include header.html %}
 
   <main>
-    <div class="post">
-      <h1>{{ page.title }}</h1>
+    <div class="post-header">
+      {% if page.header-image %}
+        <img class="post-header-image" src="{{ page.header-image | relative_url }}">
+      {% elsif page.thumbnail %}
```
Comment (jedevc) [inline _layouts/post.html]: Not a huge fan of this alternative - the thumbnails are meant to be smaller images, so adding them as a banner image will stretch them and make them look weird in a lot of cases. I'd just keep the header image property, and choose not to display the thumbnail on the post at all.

File: _layouts/post.html [hunk 2]
```diff
@@ -16,9 +16,21 @@
...
+        <img class="post-header-image" src="{{ page.thumbnail | relative_url }}">
+      {% endif %}
+    </div>
 
+    {% if page.header-image or page.thumbnail %}
+      <div class="post post-body-header-border">
```
Comment (jedevc) [inline _layouts/post.html]: I'd make the `post-body-header-border` slightly clearer, perhaps to `curved-header`. Note, we don't need to include the `post-` prefix as we're using SASS and can just use `.post.curved-header` in the SASS or whatever name you go with.

File: _layouts/post.html [hunk 3]
```diff
@@ -16,9 +16,21 @@
...
+    {% else %}
+      <div class="post">
+    {% endif %}
+      <h1 class="post-title">{{ page.title }}</h1>
```
Comment (jedevc) [inline _layouts/post.html]: Indentation here is unclear (this also applies for through the rest of the patch). For consistency for now, I'd follow the same style as used in `_layouts/page.html`.
The style there's not perfect and maybe we need to change that at some point, but probably as a different discussion under a different issue/pr.

File: css/posts.scss [hunk 4]
```diff
@@ -61,4 +70,45 @@ $preview-height: 150px;
   * {
     color: black;
   }
-}
\ No newline at end of file
+}
+
+.post-header{
+  width:100%;
```
Comment (jedevc) [inline css/posts.scss]: spacing

File: css/posts.scss [hunk 5]
```diff
@@ -61,4 +70,45 @@ $preview-height: 150px;
...
+  height: auto;
+  background: linear-gradient(to bottom, rgba(0,0,0,0) 100%, white 100%);
```
Comment (jedevc) [inline css/posts.scss]: spacing in `rgba(0,0,0,0)` should leave space after commas

File: css/posts.scss [hunk 6]
```diff
@@ -61,4 +70,45 @@ $preview-height: 150px;
...
+  margin-bottom: -5px;
+}
+
+.post-header-image {
+  max-width: 100%;
+  max-height: 50vh;
+  width: 100%;
+  height: auto;
+  object-fit:cover;
```
Comment (jedevc) [inline css/posts.scss]: spacing

File: css/posts.scss [hunk 7]
```diff
@@ -61,4 +70,45 @@ $preview-height: 150px;
...
   }
-}
\ No newline at end of file
+}
+
+.post-header{
+  width:100%;
+  height: auto;
+  background: linear-gradient(to bottom, rgba(0,0,0,0) 100%, white 100%);
+  margin-bottom: -5px;
```
Comment (jedevc) [inline css/posts.scss]: Negative marigns can cause issues later. This should probably be translated to a positive `margin-top` for the main body.

File: css/posts.scss [hunk 8]
```diff
@@ -61,4 +70,45 @@ $preview-height: 150px;
...
+
+.post-header-image {
+  max-width: 100%;
+  max-height: 50vh;
+  width: 100%;
+  height: auto;
+  object-fit:cover;
+  border-radius: 20px 20px 0px 0px;
+  z-index:-1;
+  position:relative;
```
Comment (jedevc) [inline css/posts.scss]: spacing

File: css/posts.scss (hunk 7 above)
Comment (LikkanChung) [reply css/posts.scss]: There was a weird gap in the div, but not in the image. Changing this didn't seem to fix it

**Pull request merged**